src/aleph_client/commands/instance/__init__.py

```diff
@@ -834,35 +794,50 @@ async def allocate(
 @app.command()
 async def logs(
     vm_id: str = typer.Argument(..., help="VM item hash to retrieve the logs from"),
-    domain: str = typer.Argument(..., help="CRN domain where the VM is running"),
-    private_key: Optional[str] = typer.Option(sdk_settings.PRIVATE_KEY_STRING, help=help_strings.PRIVATE_KEY),
-    private_key_file: Optional[Path] = typer.Option(sdk_settings.PRIVATE_KEY_FILE, help=help_strings.PRIVATE_KEY_FILE),
+    domain: Optional[str] = typer.Option(None, help="CRN domain on which the VM is running"),
+    private_key: Optional[str] = typer.Option(settings.PRIVATE_KEY_STRING, help=help_strings.PRIVATE_KEY),
+    private_key_file: Optional[Path] = typer.Option(settings.PRIVATE_KEY_FILE, help=help_strings.PRIVATE_KEY_FILE),
     debug: bool = False,
 ):
     """Retrieve the logs of an instance"""
     setup_logging(debug)
 
+    domain = (
+        (domain and sanitize_url(domain))
+        or await find_crn_of_vm(vm_id)
+        or Prompt.ask("URL of the CRN (Compute node) on which the instance is running")
+    )
+
     account = _load_account(private_key, private_key_file)
 
     async with VmClient(account, domain) as manager:
-        async for log in manager.get_logs(vm_id=vm_id):
-            log_data = json.loads(log)
-            if "message" in log_data:
-                echo(log_data["message"])
+        try:
+            async for log in manager.get_logs(vm_id=vm_id):
+                log_data = json.loads(log)
+                if "message" in log_data:
+                    echo(log_data["message"])
+        except:
```

Comment:
Avoid catching all exceptions in Python. What exceptions can be raised here ?

Reply:
Good point, to be fixed